sdk/go/client.go

```diff
@@ -75,9 +75,9 @@ func Connect(ctx context.Context, opts ...ClientOpt) (*Client, error) {
 	gql := errorWrappedClient{graphql.NewClient("http://"+conn.Host()+"/query", conn)}
 
 	c := &Client{
-		c:    gql,
-		conn: conn,
-		q:    querybuilder.Query(),
+		Client: gql,
+		Query:  querybuilder.Query(),
+		conn:   conn,
 	}
```

Comment:
So potentially this is controversial, so wanted to flag it :white_flag:

This changes these properties (and on all the core types) from being private to being public - this isn't *exactly* necessary for the non-module SDK, but without this, we end up having to maintain multiple variants of the codegen, one with public fields (for modules) and one with private fields (for the general case).

Personally?
- I'd rather have the module API and today's API line up as *much* as possible - even if that requires changing today's API to get there.
- I dislike needing to have these fields public at all (access to them is fine, but ideally we shouldn't be able to write to them *after* initialization). I can't think of a clever way to do this though really, but some ideas for completeness:
	- We could add new `NewDirectory`/etc calls that set private properties - but while these are "public" fields, they're not really meant for users to call.
	- We could have all of our generated types implement a custom `Setup` interface, which could then allow setting the query/client fields exactly once. We could hide the method implementation, and then just have a top-level function `Setup` (this is a variant of the above, that doesn't require a new public function for each type, just one `Setup`).
	- We could "freeze" and copy `Client` to `c` and `Query` to `q` on first use of it, then changes to the public fields don't affect private fields (so still allows initialization). 

Kinda not sure, so curious to hear if there are any issues with this. cc @vito @sipsma